docs/atomic.rst

```diff
@@ -0,0 +1,92 @@
+.. _atomic:
+
+Atomic operations
+=================
+
+In taichi, ``x[i] += 1`` is atomic but ``x[i] = x[i] + 1`` is not.
+
+For example, to perform a reduction:
+::
+    @ti.kernel
+    def sum():
+        for i in x:
+            result[None] += x[i]
+
+Or use the function ``ti.atomic_add``, which is equivalent:
+::
+    @ti.kernel
+    def sum():
+        for i in x:
+            ti.atomic_add(result[None], x[i])
+
+See https://en.wikipedia.org/wiki/Fetch-and-add for more details.
+
+
+.. note::
+    Support of atomic operation on each backends:
+
+    +------+-----------+-----------+---------+
+    | type | LLVM      | OpenGL    | Metal   |
+    +======+===========+===========+=========+
+    | i32  |    OK     |    OK     |   OK    |
+    +------+-----------+-----------+---------+
+    | f32  |    OK     |    OK     |   OK    |
+    +------+-----------+-----------+---------+
+    | i64  |    OK     |   EXT     |  MISS   |
+    +------+-----------+-----------+---------+
+    | f64  |    OK     |   EXT     |  MISS   |
+    +------+-----------+-----------+---------+
```

Comment:
And here.

Reply:
LLVM looks good. Btw should we replace `LLVM` with `CPU/CUDA`? Most users probably don't know what LLVM is.